path_login.go

```diff
@@ -78,7 +78,7 @@ func (b *backend) pathLoginUpdate(ctx context.Context, req *logical.Request, dat
 	authenticateRequestHeaders := requestHeaders.(http.Header)
 
 	// Find the targetUrl and Method
-	finalLoginPath := PathVersionBase + fmt.Sprintf(PathBaseFormat, roleName)
+	finalLoginPath := PathVersionBase + fmt.Sprintf(PathBaseFormat, "oci", roleName)
```

Comment:
I'm looking for some feedback on this part of the code. The function `requestTargetToMethodURL()` is taking in the `(request-target)` header value along with an expected HTTP method and URL path.

The function verifies that: 
1. The `(request-target)` header exists
2. The HTTP method in the header matches the expected HTTP method
3. That URL path in the header matches the expected URL path

Is there any precedent for this type of verification? I see a link to https://tools.ietf.org/html/draft-cavage-http-signatures-06#section-2.3 in the CLI code.

Reply:
This is problematic for this PR as we need to know the mount path segment in `pathLoginUpdate()` to do this verification. Using `"oci"` as currently in this PR won't work.

Reply:
@vuhphamw Do you have any advice here?

Reply:
@austingebauer OCI REST API signature signing process requires the header includes (request-target) as described here https://docs.cloud.oracle.com/en-us/iaas/Content/API/Concepts/signingrequests.htm#five

In order to support mount path, we need to somehow pass the mount path to this function and validate the URL path bases on that, i.e. `/auth/mounted-path/login/devRole`. Is there a way to pass the mount path to this method via context or something else?

Reply:
Thanks, @vuhphamw. Makes sense that the header needs to exist in order to comply with OCI API request signing. I don't think we can use a context here. The Vault CLI and other clients (e.g., curl) exercise this API over HTTP.

Is there any specific reason we're validating the value of the `(request-target)` from the [POST body](https://www.vaultproject.io/api-docs/auth/oci#sample-payload-2) in `pathLoginUpdate()` instead of just letting OCI API verify the signature? It seems we're currently just checking the `(request-target)` value against some constants ([path_login.go#L19](https://github.com/hashicorp/vault-plugin-auth-oci/blob/master/path_login.go#L19)) in the backend before sending them to the OCI API. Would checking that the `(request-target)` value in the POST body be non-empty or a regex suffice?

Reply:
@austingebauer Yes, I think regex is sufficient. Please ensure the target path match `/auth/%s/login/%s`. I believe it was just extra validation to ensure the header has right fields for signing process.

Reply:
@vuhphamw I've updated this PR to use a regex to do the validation. Please have a look and let me know your thoughts. 

I kept the actual login role in the regex. The only part of the URL path which has an open pattern is the mount path, which uses the ASCII character class `[[:ascii:]]`. I'm using that character class because of the limitation noted in the [/sys/mounts](https://www.vaultproject.io/api-docs/system/mounts#path) documentation.